src/main/java/com/minecolonies/coremod/colony/managers/GraveManager.java

```diff
@@ -197,7 +197,7 @@ public boolean reserveGrave(@NotNull final BlockPos pos)
     @Override
     public void unReserveGrave(@NotNull final BlockPos pos)
     {
-        if (graves.get(pos))
+        if (graves.containsKey(pos) && graves.get(pos) != null)
```

Comment:
Doesn't need the null check, contains does this already, so this should be graves.containsKey(pos) && graves.get(pos)
The method that calls this then needs the te instanceof Grave instead of != null too

Reply:
Ok, done.

Reply:
doing a containsKey followed by get will perform the same operation twice on the map. It would be cheaper to just use get and check the result if you never insert null values.

Reply:
you're right that it would be cheaper, although this place isnt called frequent enough to make a difference and is a bit nicer to read this way